api/api/serializers/media_serializers.py

```diff
@@ -71,11 +70,13 @@ def validate_page_size(self, value):
         if is_anonymous:
             try:
                 validator(value)
-            except ValidationError as e:
-                raise NotAuthenticated(
-                    detail=e.message,
-                    code=e.code,
-                )
+            except Exception as e:
+                # TODO: figure out why except ValidationError doesn't catch this
+                if type(e).__name__ == "ValidationError":
```

Comment:
Is this todo still valid?

Reply:
Yes, it's still valid. I could not figure out why this didn't work:
```py
import rest_framework.exception.ValidationError
catch ValidationError
```
Even though the name of this error is `ValidationError`.

I need help with this

Reply:
It's probably the Django rather than DRF class. The DRF class does not inherit from the Django classes:

https://github.com/encode/django-rest-framework/blob/master/rest_framework/exceptions.py#L143

https://github.com/django/django/blob/main/django/core/exceptions.py#L134

Reply:
DRF has the same difficulty itself, it turns out. Here they use `except (ValidationError, DjangoValidationError)` as a "catch all" for the two potential approaches: https://github.com/encode/django-rest-framework/blob/master/rest_framework/serializers.py#L453

Reply:
Thank you, @sarayourfriend! I updated the code and removed the `TODO` comment.

One thing I'm not sure about in this PR is whether the tests do all of the checks.
This is a list of checks that Schemathesis can perform: [`not_a_server_error`, `status_code_conformance`, `content_type_conformance`, `response_headers_conformance`, `response_schema_conformance`]. The Pytest command we use is supposed to do all of the checks by default [^1]. 

In the first iteration of this PR, I opened 2 PRs. This one only added the test for response status codes, and #3447 added the changes for checking the `response_schema_conformance` and other checks. There, for instance, I had to add a new serializer for audio alt_files to make the responses conform the schema. Here, we don't seem to have such a problem. Is it because something changed in the responses so that they are now conformant, or because the check is not run correctly?

[^1]: https://schemathesis.readthedocs.io/en/stable/python.html#how-are-responses-checked

Reply:
The spec in this branch has `any or null` for alt files. That matches production: https://api.openverse.engineering/v1/#tag/audio/operation/audio_detail

I didn't run into that issue when I was working on the changes I contributed to this PR, and I don't know how I would reproduce it. Based on the schema in main on this PR, I can't think of any reason schemathesis would ever error on any value at all in `alt_files`, unless there was a version of this PR that changed the schema type for that field away from `any or null`.

The schema type checks are working fine, as far as I can tell. The changes to the stats endpoint documentation are proof of that (schemathesis will fail if you remove the `many=True` because without that the schema does not wrap the response in a list, which the endpoint always returns).

Reply:
Thank you! I didn't realize that `audio_detail` is set to `any or null`. I added the `alt_files` serializer from the other PR to make the types more accurate.
I didn't realize that you've also added the `many=True` to this PR. Those were also causing the response shape mismatch error.